### PR TITLE
EOS-13759 [LDR-R1 SW upgrade] s3 config side

### PR DESCRIPTION
**Pull request by abhilekh268**

### CODE REVIEW

File: server/s3_option.cc [hunk 1]
```diff
@@ -602,6 +604,8 @@ bool S3Option::load_section(std::string section_name,
              &libevent_pool_buffer_size);
       sscanf(libevent_max_read_size_str.c_str(), "%zu",
              &libevent_max_read_size);
+    } else if (section_name == "S3_VERSION_CONFIG") {
+      s3_version = s3_option_node["VERSION"].as<std::string>();
```
Comment (knrajnambiar76) [inline server/s3_option.cc:608]: This field cant be empty, it has to be there and it its empty then give error. We can take default version

File: s3config.release.yaml [hunk 2]
```diff
@@ -107,3 +107,5 @@ S3_THIRDPARTY_CONFIG:
    S3_LIBEVENT_POOL_RESERVE_SIZE: 134217728             # 128MB, deny PUT request if mempool free space is less than the mentioned size in bytes
    S3_LIBEVENT_MEMPOOL_ZERO_BUFFER: false               # Enable Libevent Mempool 'zeroing' after use (like secure erase) - disabled by default
    S3_LIBEVENT_POOL_RESERVE_PERCENT: 5                  # 5%, deny PUT request if mempool free space in percent is less than the mentioned percent
+S3_VERSION_CONFIG:
+   VERSION: 1.0.0
```
Comment (shalakadharap) [inline s3config.release.yaml:111]: Please keep version as 1 everywhere
Confirmed with @nileshgovande

**Review by shalakadharap — CHANGES_REQUESTED** [review]

File: server/s3_option.cc (hunk 1 above)
Comment (abhilekh268) [reply server/s3_option.cc:608]: yes default is specified.

File: s3config.yaml [hunk 3]
```diff
@@ -107,3 +107,5 @@ S3_THIRDPARTY_CONFIG:
    S3_LIBEVENT_POOL_RESERVE_SIZE: 1048576               # Deny PUT request if mempool free space is less than the mentioned size in bytes
    S3_LIBEVENT_MEMPOOL_ZERO_BUFFER: false               # Enable Libevent Mempool 'zeroing' after use (like secure erase) - disabled by default
    S3_LIBEVENT_POOL_RESERVE_PERCENT: 5                  # Deny PUT request if mempool free space in percent is less than the mentioned percent
+S3_VERSION_CONFIG:
+   VERSION: 1.0.0
```
[the remark this answers is not in the document]
Comment (abhilekh268) [reply s3config.yaml]: yes it's taken as string only.

File: s3config.release.yaml (hunk 2 above)
Comment (abhilekh268) [reply s3config.release.yaml]: done

**Review by shanbhagsv1 — APPROVED** [review]

**Review by shalakadharap — APPROVED** [review]

**Comment by abhilekh268** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/51/ jenkins job

**Pull request merged**